### DISCUSSION TITLE
Configuring SLIP TAP as the Network Driver on NXP RT1060 EVK

### DISCUSSION THREAD

**Discussion by ofirshe** [zephyrproject-rtos/zephyr#82411] [Q&A] [answered]
Hey,

I am running a TCP server application on the NXP RT1060EVB microcontroller. While networking will also be available on our future board, I would like to implement an alternative communication method in case the NXP Ethernet (NXP ENET) interface fails to initialize smoothly.

I noticed that Zephyr supports SLIP TAP, and I aim to configure the environment on the RT1060EVB to use SLIP as the network driver instead of the default Ethernet path. I attempted the following configurations:
```
CONFIG_NETWORKING=y  
CONFIG_NET_UDP=y  
CONFIG_NET_TCP=y  
CONFIG_NET_IPV4=y  
CONFIG_INIT_STACKS=y  
CONFIG_NET_LOG=y  
CONFIG_NET_CONFIG_SETTINGS=y  
CONFIG_NET_CONFIG_NEED_IPV4=y  
CONFIG_NET_CONFIG_MY_IPV4_ADDR="172.18.133.10"  
CONFIG_NET_CONFIG_MY_IPV4_NETMASK="255.255.240.0"  
CONFIG_NET_CONFIG_MY_IPV4_GW="172.18.143.254"  
CONFIG_NET_L2_ETHERNET=y  
CONFIG_NET_CONFIG_INIT_TIMEOUT=0  
CONFIG_NET_CONTEXT_RCVTIMEO=y  
CONFIG_NET_SOCKETS=y  
CONFIG_POSIX_API=y  
CONFIG_NET_TX_STACK_SIZE=8192  
CONFIG_NET_RX_STACK_SIZE=8192  
CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=8192  
CONFIG_NET_PKT_RX_COUNT=64  
CONFIG_NET_PKT_TX_COUNT=64  
CONFIG_NET_BUF_RX_COUNT=64  
CONFIG_NET_BUF_TX_COUNT=64  
CONFIG_NET_MAX_CONN=64  
CONFIG_NET_MAX_CONTEXTS=64  
# Switch to SLIP mode
CONFIG_ETH_NXP_ENET=n  
CONFIG_NET_DRIVERS=y  
CONFIG_SLIP=y  
CONFIG_SLIP_TAP=y  
```

However, when compiling the project using `west build -b mimxrt1060_evk`, the `build/zephyr/.config` file still shows `CONFIG_ETH_NXP_ENET=y`. Additionally, the build includes references to the NXP Ethernet driver, PHY, and SLIP TAP in the `build/zephyr/drivers/ethernet` directory.
<img width="483" alt="image" src="https://github.com/user-attachments/assets/cd83b6d8-1572-49dd-83c8-7b201adbb32e">

I would appreciate any guidance on how to correctly configure the environment to use SLIP TAP as the sole network driver on the RT1060EVB.

Thank you for your assistance.

Best regards,
Ofir.

**Comment by jukkar** [comment]
Using SLIP outside of Qemu is not really tested but should work as long as there is a UART that is providing the connection.
Not sure what selects the `CONFIG_ETH_NXP_ENET` symbol, you could use menuconfig to see the dependencies for that symbol.

**Comment by dpiskula-nxp** [comment]
Hello @ofirshe, as for the ENET drivers getting included, I think the issue in your case is _CONFIG_NET_L2_ETHERNET=y_. This config is enabled by default when you set _CONFIG_NETWORKING=y_ and you also configure it yourself (on the line right below the IP address configurations). Try setting it to n.

It's also selected by [CONFIG_NET_SLIP_TAP](https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_NET_SLIP_TAP) if you enable that, so keep that in mind.

**Reply by ofirshe** [reply]
Hey @dpiskula-nxp, 

First, I don’t enable `CONFIG_NET_SLIP_TAP` since it’s meant for the QEMU target.
<img width="631" alt="image" src="https://github.com/user-attachments/assets/075faada-7edc-4ec3-8219-94f15ef904d5" />
Instead, I only enable `CONFIG_SLIP` and `CONFIG_SLIP_TAP`.

As for `CONFIG_NET_L2_ETHERNET`, it’s selected by `CONFIG_SLIP_TAP`, so disabling it isn’t the solution.
<img width="408" alt="image" src="https://github.com/user-attachments/assets/db4ae16c-bac9-4d86-9360-0e5fa2fa6c7a" />

**Comment by jukkar** [comment]
@ofirshe Just wondered did you get the SLIP working outside of Qemu?

**Reply by ofirshe** [reply]
Hi @jukkar,
I'm currently setting up SLIP over LPUART3 on the RT1060-EVK. I'll provide an update once I have the environment fully configured and can test for any issues. Since I'm using Windows, integrating SLIP over one of the available COM ports isn't quite straightforward.

**Comment by ofirshe** [comment]
Hi @jukkar,

I have set up an Ubuntu VM on my Windows machine and connected it via a serial interface to the RT1060-EVK board. On my PC, the connection is mapped to `COM3`, which corresponds to `/dev/ttyS2` on the VM.

This is my `prj.conf` file:
```
CONFIG_NETWORKING=y
CONFIG_NET_UDP=y
CONFIG_NET_TCP=y
CONFIG_NET_IPV4=y
CONFIG_INIT_STACKS=y
CONFIG_NET_LOG=y
CONFIG_NET_CONFIG_SETTINGS=y
CONFIG_NET_CONFIG_NEED_IPV4=y
CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.0.2.1"
CONFIG_NET_L2_ETHERNET=y
CONFIG_NET_CONFIG_INIT_TIMEOUT=0
CONFIG_NET_CONTEXT_RCVTIMEO=y
CONFIG_NET_SOCKETS=y
CONFIG_POSIX_API=y
CONFIG_NET_TX_STACK_SIZE=8192
CONFIG_NET_RX_STACK_SIZE=8192
CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=8192
CONFIG_NET_PKT_RX_COUNT=64
CONFIG_NET_PKT_TX_COUNT=64
CONFIG_NET_BUF_RX_COUNT=64
CONFIG_NET_BUF_TX_COUNT=64
CONFIG_NET_MAX_CONN=64
CONFIG_NET_MAX_CONTEXTS=64
# Switch to SLIP mode
CONFIG_NET_DRIVERS=y  
CONFIG_SLIP=y
CONFIG_SLIP_TAP=y
```

On the board, I configured `CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.0.2.1"`. On the VM, I set up SLIP with the following commands:
`sudo slattach -v -s 115200 -p cslip -L /dev/ttyS2`

I then configured the SLIP network interface as follows:
```
sudo ifconfig sl0 192.0.2.2 netmask 255.255.255.0 up
sudo route add -net 192.0.2.0 netmask 255.255.255.0 dev sl0
```

I'm sending a ping to the board, but I'm not receiving any response. Could there be an issue with the configurations?

Thanks!

**Reply by ofirshe** [reply, marked as answer]
I got it working with the help of @krish2718 [here](https://github.com/zephyrproject-rtos/zephyr/discussions/48697#discussioncomment-11642727).
